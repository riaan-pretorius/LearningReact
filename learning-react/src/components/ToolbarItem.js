import React, { Component } from 'react';
import { NavItem } from 'reactstrap';
import {
    BrowserRouter as Router,
    Route,
    NavLink
  } from 'react-router-dom'


class ToolbarItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            to: props.to,
            text: props.text
        }
    }
    render() {
        return (
            <NavItem>
                <NavLink to={this.state.to} className="nav-link" exact>{this.state.text}</NavLink>
            </NavItem>
        )
    }
}

export default ToolbarItem