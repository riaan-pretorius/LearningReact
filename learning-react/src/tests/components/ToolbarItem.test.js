import React from 'react'
import ToolbarItem from '../../components/ToolbarItem'
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'
import {
  MemoryRouter
} from 'react-router-dom'

test('ToolbarItem Test Snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <ToolbarItem to="/" text="Testing" />
    </MemoryRouter>
  )
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
})

test('ToolbarItem Test', () => {
    const component = mount(
      <MemoryRouter>
        <ToolbarItem to="/" text="Testing" />
      </MemoryRouter>
    )
    
    let a = component.find('a')
    expect(a.exists()).toBeTruthy()
    expect(a.text()).toBe("Testing")
    expect(a.prop('href')).toBe("/")
})