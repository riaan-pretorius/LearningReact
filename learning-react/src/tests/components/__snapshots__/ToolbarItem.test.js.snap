// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarItem Test Snapshot 1`] = `
<li
  className="nav-item"
>
  <a
    aria-current="true"
    className="nav-link active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Testing
  </a>
</li>
`;
